src/js/trove/iolib.js

```diff
@@ -0,0 +1,51 @@
+({
+    requires: [],
+    provides: {
+      values: {
+        "prompt": "tany"
+      }
+    },
+    nativeRequires: ["fs"],
+    theModule: function(RUNTIME, NAMESPACE, uri, fs) {
+      // https://github.com/nodejs/node/issues/28243#issuecomment-502402453
+      function PromptIO(msg) {
+        // console.log(RUNTIME);
+        fs.writeSync(RUNTIME.stdout_fd, msg);
+        let s = '';
+        while (true) {
+          let buf = Buffer.alloc(1);
+
+          // TODO: figure out why this needs to be within a try-catch
+          try {
+            // TODO: make sure this line works on non-unix OS
+            fs.readSync(RUNTIME.stdin_fd, buf, 0, 1, null);
+          } catch (e) {
+            if (e.code !== "EAGAIN") {
+              throw e;
+            }
+          }
+
+          // TODO: elaborate on more exit conditions
+          if (buf.toString("utf-8") === "\n") {
```

Comment:
Do I understand this code correctly as saying that a user has to type `hello<enter><enter>` in order to actually return from `prompt`?  That seems...unintuitive.